project/src/main/java/com/google/sps/servlets/RegistrationServlet.java

```diff
@@ -0,0 +1,64 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.import java.io.IOException;
+
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.Optional;
+import com.google.sps.data.DataAccessor;
+import com.google.sps.data.UserVerifier;
+import com.google.common.annotations.VisibleForTesting;
+
+ /**
+ * A servlet that registering a user, according to a given token.
+*/
+@WebServlet("/register")
+public final class RegistrationServlet extends HttpServlet {
+
+  private static final long serialVersionUID = 1L;
+  private UserVerifier userVerifier;
+  private DataAccessor dataAccessor;
+
+  @Override
+  public void init() {
+    this.userVerifier = UserVerifier.create("CLIENT_ID");
+    this.dataAccessor = new DataAccessor();
+  }
+
+  @VisibleForTesting
+  void init(UserVerifier userVerifier, DataAccessor dataAccessor) {
+    this.userVerifier = userVerifier;
+    this.dataAccessor = dataAccessor;
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String userIdToken = request.getParameter("idToken");
+    if (userIdToken == null) {
+      return;
```

Comment:
If the IdToken passed to the getUserIdByToken is never null, I think the null test in the verifier might be redundent (or, this check is redundent since this null situation is handeled in the verifier).

Reply:
We want to make this code bullet-proof to any changes someone might do with our code. This makes sure that even if someone changes the way the servlet is called, or calls the servlet from a new place in the code, it won't cause errors (in this case - NullPointerException). 
(Not to mention that this is a scenario that we want to test, so the function must behave this way...)

Reply:
Yes I agree that it's good that the verifier handels getting null, just thought that since the verifier handels the null case then this first if statement isn't needed, but I understand you want that double check.

Reply:
Maybe add an error code to the response, marking that you got an invalid argument?

Reply:
great idea, done